pkg/block/indexheader/lazy_binary_reader_test.go

```diff
@@ -52,27 +64,38 @@ func TestNewLazyBinaryReader_ShouldBuildIndexHeaderFromBucket(t *testing.T) {
 	testutil.Ok(t, err)
 	testutil.Ok(t, block.Upload(ctx, log.NewNopLogger(), bkt, filepath.Join(tmpDir, blockID.String()), metadata.NoneFunc))
 
-	m := NewLazyBinaryReaderMetrics(nil)
-	bm := NewBinaryReaderMetrics(nil)
-	r, err := NewLazyBinaryReader(ctx, log.NewNopLogger(), bkt, tmpDir, blockID, 3, m, bm, nil)
-	testutil.Ok(t, err)
-	testutil.Assert(t, r.reader == nil)
-	testutil.Equals(t, float64(0), promtestutil.ToFloat64(m.loadCount))
-	testutil.Equals(t, float64(0), promtestutil.ToFloat64(m.unloadCount))
-
-	// Should lazy load the index upon first usage.
-	v, err := r.IndexVersion()
-	testutil.Ok(t, err)
-	testutil.Equals(t, 2, v)
-	testutil.Assert(t, r.reader != nil)
-	testutil.Equals(t, float64(1), promtestutil.ToFloat64(m.loadCount))
-	testutil.Equals(t, float64(0), promtestutil.ToFloat64(m.unloadCount))
-
-	labelNames, err := r.LabelNames()
-	testutil.Ok(t, err)
-	testutil.Equals(t, []string{"a"}, labelNames)
-	testutil.Equals(t, float64(1), promtestutil.ToFloat64(m.loadCount))
-	testutil.Equals(t, float64(0), promtestutil.ToFloat64(m.unloadCount))
+	for _, lazyDownload := range []bool{true} {
```

Comment:
should be `false,true`, right?

Reply:
Can we run those inner tests as `t.Run("..", func(tt *testing.T) {...})` ?

Reply:
Addressed